### PR TITLE
fix the back to update the shipping address title

### PR DESCRIPTION
**Pull request by vikas-cis-zz**
https://particl.atlassian.net/browse/PD-469

### CODE REVIEW

File: src/app/market/buy/checkout-process/checkout-process.component.ts [hunk 1]
```diff
@@ -214,8 +214,11 @@ export class CheckoutProcessComponent implements OnInit, OnDestroy {
     })
   }
 
-  select(address: Address) {
+  select(address: Address, isNewProfile?: boolean) {
     this.log.d('Selecting address with id: ' + address.id);
+    if (isNewProfile) {
+      this.profile.shippingAddresses = [address];
```
Comment (zaSmilingIdiot) [inline src/app/market/buy/checkout-process/checkout-process.component.ts]: When navigating from step 3 to 2 int he checkout process, the shipping profile is now correctly selected, if one was previously selected. However, it is the only option in the shipping profile dropdown, which I suspect is a consequence of this specific line of code.

**Review by zaSmilingIdiot — CHANGES_REQUESTED** [review]
Shipping addresses appear to be obtained only once, during the initialisation of the form. Specifically, when the following line is executed:
`this.getProfile();`

The shipping addresses should probably be updated and a shipping profile address re-selected if one was previously selected, when the user navigates back from step 3 to step 2 of the checkout process.

This PR also seems to be missing the change to return to step 1 of the checkout process when the user navigates to a different page and back.

**Comment by vikas-cis-zz** [comment]
@zaSmilingIdiot it is still in progress so obviously the second part hasn't been pushed yet :P :D 

Thanks

**Comment by vikas-cis-zz** [comment]
@zaSmilingIdiot yeah now you can check the branch as ready for review :)

Thanks

**Comment by zaSmilingIdiot** [comment]
@vikas-cis Pushing the newly created shipping profile into the array of shipping profiles does not consider a number of scenarios, such as:
- if the shipping profile is already in the array (if the user has navigated back and forth between steps 2 and 3 multiple times),
- if the the address is being updated instead of a new address being created.

There are a couple more scenarios to consider.
Nevertheless, I've updated the code to reflect what I think is the correct way to handle all the scenarios I can think of. Essential is that:
a. if the user has selected a saved address AND modified the address but NOT requested the address to be saved: ensure that the previously selected address is populated in the dropdown and the modified form values are displayed with their modifications when returning to step 2;
b. if the user has selected a saved address AND modified the address but HAS requested the address to be saved: has the same result as (a), but with the address having been updated (this includes the saved address name)
c. if the user has NOT selected a saved address: the same results from (b) should apply but the dropdown should be empty when returning to step 2.

For sanity checking, can you please verify that this holds and that nothing else has broken?

Thanks.

**Comment by vikas-cis-zz** [comment]
@zaSmilingIdiot yeah I gotch you :)

Thanks

**Comment by vikas-cis-zz** [comment]
@zaSmilingIdiot  yeah I did look and seems working well to me 👍 

Thanks

**Review by zaSmilingIdiot — APPROVED** [review]

**Comment by zaSmilingIdiot** [comment]
Fixes https://github.com/particl/particl-desktop/issues/1198

**Pull request merged**